Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl

```diff
@@ -32,7 +32,12 @@ namespace AZ
     template <typename BASE_TYPE, ThreadSafety THREAD_SAFETY>
     inline void ConsoleDataWrapper<BASE_TYPE, THREAD_SAFETY>::operator =(const BASE_TYPE& rhs)
     {
+        const BASE_TYPE currentValue = this->m_value;
         this->m_value = rhs;
+        if(currentValue != rhs)
+        {
+            InvokeCallback();
+        }
```

Comment:
We can save the assignment until after the comparison check to validate that the value is different
```suggestion
        if(currentValue != rhs)
        {
            this->m_value = rhs;
            InvokeCallback();
        }
```

Reply:
I didn't do it that way because I wanted to avoid any accidental side effects if the cvar wrapped typed overrides the `operator=`

Reply:
You mean `operator!=`.
That function is just for comparisons and should be const.
Plus the only types we support in CVars are `AZStd::fixed_string`, AZ::Vector*`, the arithmetic and floating point builtin types.

Reply:
No, I meant `operator=` 
On this line `this->m_value = rhs;` the type of m_value comes from the creator of the cvar and can be any type that supports the basic operators, so it doesn't seem wise to assume that they would want the call to `operator=` optimized away if we believe the value has not changed.

Reply:
Ok that sounds fine then.
In case they have an `operator=` where they cache a value in the assignment.

Well there is one more nitpicky change to make then.
Can you add a space after the `if` and before the `(` then?

Reply:
I will add the space.